### PR TITLE
[lc_ctrl] Add cSHAKE128 token hashing to the LC scripts 

### PR DESCRIPTION
**Pull request by msfschaffner**
This switches to the development version of pycryptodome (see #6694), and uses the new cSHAKE library function in the life cycle generation scripts to precompute the design constants.

This addresses #6548.

### CODE REVIEW

**Comment by msfschaffner** [comment]
CC @moidx @cdgori

**Review by cindychip — APPROVED** [review]
LGTM

File: util/design/lib/LcStEnc.py [hunk 1]
```diff
@@ -181,14 +182,24 @@ def _validate_tokens(config):
     '''Validates and hashes the tokens'''
     config.setdefault('token_size', 128)
     config['token_size'] = check_int(config['token_size'])
+    # This needs to be byte aligned
+    if config['token_size'] % 8:
+        raise RuntimeError('Size of token {} must be byte aligned'
```
Comment (imphil) [inline util/design/lib/LcStEnc.py:187]:
```suggestion
        raise ValueError('Size of token {} must be byte aligned'
```

Would be more typical, but doesn't really matter here.

**Review by imphil — APPROVED** [review]
LGTM for the Python part.

File: util/design/lib/LcStEnc.py (hunk 1 above)
Comment (msfschaffner) [reply util/design/lib/LcStEnc.py]: Thanks, I've changed this to `ValueError`

**Review by cdgori — COMMENTED** [review]
This looks as expected.  Thanks.

**Pull request merged**